src/Middleware/RequestQueue/test/SemaphoreWrapperTests.cs

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Xunit;
+using System.Threading;
+using System.Threading.Tasks;
+using System;
+using System.Runtime.CompilerServices;
+using Microsoft.AspNetCore.Testing;
+
+namespace Microsoft.AspNetCore.RequestQueue.Tests
+{
+    public class SemaphoreWrapperTests
+    {
+        [Fact]
+        public async Task TestBehavior()
```

Comment:
I'd rename some of these tests to better describe the behavior they are testing. For example, you could rename this test to `EnterAndLeaveChangesSpotsLeftCorrectly` or something to that effect.